pyomo/contrib/pynumero/interfaces/cyipopt_interface.py

```diff
@@ -328,24 +329,44 @@ def scaling_factors(self):
         return obj_scaling, x_scaling, g_scaling
 
     def objective(self, x):
-        self._set_primals_if_necessary(x)
-        return self._nlp.evaluate_objective()
+        try:
+            self._set_primals_if_necessary(x)
+            return self._nlp.evaluate_objective()
+        except PyNumeroEvaluationError:
+            # TODO: halt_on_evaluation_error option. If set, we re-raise the
+            # original exception.
+            raise cyipopt.CyIpoptEvaluationError(
```

Comment:
This requires cyipopt 1.3.0.  Should we update the `attempt_import` only allow cyipopt >= 1.3.0, or should we update the importer on line 28 to define a `cyipopt.CyIpoptEvaluationError` exception as a compatibility shim for older versions?  

I would tend to lean toward the latter.

Reply:
Good point. I agree with defining a dummy evaluation error for compatibility.

This also raises the question of behavior consistency among different CyIpopt versions. Should the default be to re-raise the original exception, with a `halt_on_evaluation_error` flag that a user can set to `False` if they want to use the new feature?

Reply:
This is in contrib, so backwards compatibility requirements are not as strict. I think I would vote for consistency with the ASL IPOPT interface and have `halt_on_evaluation_error` default to False.